frontend/src/routers/index.ts

```diff
@@ -10,6 +10,12 @@ router.beforeEach((to, from, next) => {
     axiosCanceler.removeAllPending();
     const globalStore = GlobalStore();
 
+    if (globalStore.isIntl && to.path.includes('xpack')) {
+        next({ name: '404' });
+        NProgress.done();
+        return;
+    }
+
     if (to.name !== 'entrance' && !globalStore.isLogin) {
         next({
             name: 'entrance',
```

Comment:
There are no apparent errors or issues with this code. However, there is an opportunity for further optimization:

1. **Code DRYing**: The `next({ name: '404' });` line appears twice in the same branch. Consider extracting it into a separate function call to make the code cleaner.

Here's how you can optimize it:

```javascript
@@ -10,6 +10,9 @@ router.beforeEach((to, from, next) => {
   axiosCanceler.removeAllPending();
   const globalStore = GlobalStore();

+  handleUnauthorizedRoute(next);
+
   if (!globalStore.isLogin && to.name !== 'entrance') {
       next({
           name: 'entrance',
```

Create this helper function:

```javascript
function handleUnauthorizedRoute(next) {

    nprogress.done()

    if(to.path.includes('xpack')){
        return 
    }

    console.log("Unauthorized path")
    
    // You may want to log more specific information here, like the current route
}
```

This way, the duplicate code block is removed and organized under its own named function, improving readability and maintainability.